packages/db/src/meta/interface.ts

```diff
@@ -69,12 +88,27 @@ export const forAttachAndSchema = <C extends Collections>(options: {
     };
   };
 
-  const serve = (config: TruffleConfig | ConnectOptions<C>) => {
-    const options =
-      "working_directory" in config // TruffleConfig case
-        ? toConnectOptions(config)
-        : (config as ConnectOptions<C>);
-
+  const serve = (config: TruffleConfig | ConnectOptions<C> | undefined) => {
+    // TODO: get rid of @truffle/config stuff and rename field to "directory"
+    let options;
+    if (config && "working_directory" in config) {
+      // TruffleConfig case
+      options = toConnectOptions(config as TruffleConfig);
+    } else if (config && "directory" in config) {
+      // ConnectOptions case
+      options = config;
+    } else {
+      const configStore = new Configstore(
+        "truffle",
+        {},
+        { globalConfigPath: true }
+      );
+      const truffleDataDirectory = path.dirname(configStore.path);
+      options = {
+        directory: truffleDataDirectory,
+        adapter: (config || {}).adapter
+      };
+    }
```

Comment:
On the topic of getting rid of `directory` from `ConnectOptions`, that would mean we won't be able to have this logic here. We'll have to find another place to do this.

Reply:
Why would we get rid of `directory` from `ConnectOptions`? We'll still need to allow the user to input a directory if they want to.

Reply:
resolved